### DISCUSSION TITLE
How to do properly an optimization with constraints?

### DISCUSSION THREAD

**Discussion by ouielba90** [grimme-lab/xtb#823] [Q&A]
I have a molecule that is in an SDF format and I want to optimize it using constraints. I have added the following lines in the SDF fine and then I simply run `xtb molecule.sdf --opt`,
```
$constraint
dihedral: 23, 24, 25, 7, 0
$end
```

But the program ignores it. I also tried to put these lines in an input file (input.inp) and then run `xtb molecule.sdf --input input.inp`, yet I got the same issue.

What am I doing wrong?

**Comment by TyBalduf** [comment]
I'm not sure of your first command, do you mean you put `$constrain ...` on the commandline?

If so, I believe you just need to do `xtb molecule.sdf --opt --input input.inp`. The xcontrol input sets input parameters, but doesn't turn on particular calculation types, i.e you still need to specify `opt` to get it to start an optimization and then the setting a constraint will constrain that bond/angle/dihedral during the opt.
